semgrep-core/matching/Generic_vs_generic.ml

```diff
@@ -995,7 +996,17 @@ and m_list__m_argument (xsa: A.argument list) (xsb: A.argument list) =
   (* dots: ..., can also match no argument *)
   | [A.Arg (A.Ellipsis _i)], [] ->
       return ()
-
+  (* Fix to issue 1173. Case where there are ellipsis to the left and right of a statement.
+    * Order should not matter in this case, as we can see from the go structs example.
+    * However, there is an issue where this will match a correct line twice occasionally.
+    * Look into m_list_in_any_order for correction in this case. *)
+   | (A.Arg (A.Ellipsis _i))::xsa, xsb
+     when let rev_xsa = rev xsa in
```

Comment:
Is this trying to get the last element in the list?

If so, I think you can do `List.nth xsa (List.length xsa - 1)`.

I'm also wondering what happens with ellipses in the middle of the list; e.g. `..., a, ..., b, ...`. I'm wondering if we need some sort of recursive solution here? (cc. @aryx )

Reply:
There's also Common2.list_last.
As I said, improving the arguments matching for Composite literals is not the right approach.
We already have code to handle matching elements in any order for Dict, fields, etc. We just need
to generate something better in go_to_generic.ml from CompositeLit.

Reply:
Whatever you all do, please avoid exceptions when you can (unless you do want a fatal error): if you want the last element of a list, such function should return an option because the list could be empty. Similarly, don't use `List.hd`, use pattern matching or a function that returns an option, like:
```ocaml
let first = function [] -> None | x :: _ -> Some x
```

Reply:
done! thanks for the comments!